src/main/java/org/opentestsystem/shared/contentspecid/legacy/LegacyContentSpecIdConverter.java

```diff
@@ -73,147 +59,145 @@ public ContentSpecFormat getFormatType(@Nonnull final String input) throws IdPar
-                id.setTarget(convertTarget(target));
-
                 if (parts.length > 2) {
-                    id.setCcss(convertCcss(parts[2]));
+                    builder.ccss(parts[2]);
                 }
 
-                id.setDomain(convertDomain(domain, getLegacyDomain(id), id.deriveDomain(),
-                        false, id.getGrade(), id.getCcss()));
                 break;
 
             case MATH_V4:
             case MATH_V5:
-                target = "";
-                parseGrade = null;
+                if (parts.length > 0) {
+                    builder.claim(parts[0]);
+                }
+
+                if (parts.length > 1) {
+                    builder.domain(parts[1]);
+                }
 
                 if (parts.length > 2) {
                     int dash = parts[2].indexOf('-');
                     if (dash >= 0) {
-                        target = parts[2].substring(0, dash);
-                        parseGrade = parts[2].substring(dash + 1);
+                        builder.target(parts[2].substring(0, dash));
                     } else {
-                        target = parts[2];
+                        builder.target(parts[2]);
                     }
                 }
 
-                if (parseGrade != null) {
-                    id.setGrade(convertGrade(parseGrade));
-                } else {
-                    id.setGrade(defaultGrade);
+                if (parts.length > 3) {
+                    builder.emphasis(parts[3]);
                 }
 
-                if (parts.length > 0) {
-                    id.setClaim(convertClaim(parts[0]));
+                if (parts.length > 4) {
+                    builder.ccss(parts[4]);
                 }
 
-                id.setTarget(convertTarget(target));
+                break;
 
-                if (parts.length > 4) {
-                    id.setCcss(convertCcss(parts[4]));
+            case MATH_V6:
+                String claim = "";
+                if (parts.length > 0) {
+                    claim = parts[0];
+                    builder.claim(claim);
                 }
 
                 if (parts.length > 1) {
-                    boolean isMathNonPrimaryClaim = (id.getClaim() != ContentSpecClaim.C1);
-                    id.setDomain(convertDomain(parts[1], getLegacyDomain(id), id.deriveDomain(),
-                            isMathNonPrimaryClaim, id.getGrade(), id.getCcss()));
-                }
-                if (parts.length > 3) {
-                    validateEmphasis(parts[3], id.getEmphasis());
+                    if (claim.equals("1")) {
+                        builder.contentCategory(parts[1]);
+                    } else {
+                        builder.domain(parts[1]);
+                    }
                 }
-                break;
 
-            case MATH_V6:
-                target = "";
-                parseGrade = null;
+//                if (parts.length > 2) {
```

Comment:
Is this something we can finish or does the original library have this commented?